File: 2486.py
while True:
    casos = int(input())
    if casos == 0:
        break
    mg = 0
    for i in range(0, casos):
        n = input()
        numero = int(n[:n.index(" ")])
        fruta = n[n.index(" ")+1:]
        match fruta:
            case "suco de laranja":
                mg += numero*120

            case "morango fresco":
                mg += numero*85

            case "mamao":
                mg += numero*85

            case "goiaba vermelha":
                mg += numero*70

            case "manga":
                mg += numero*56

            case "laranja":
                mg += numero*50

            case "brocolis":
                mg += numero*34
    if mg > 130:
        print(f"Menos {mg-130} mg")
    elif mg >= 110:
        print(f"{mg} mg")
    else:
        print(f"Mais {110-mg} mg")

File: 2486.cpp
#include <iostream>
#include <string>
using namespace std;

int main(){
    int numero_casos, mg, numero, pos;
    string frase, fruta;
    while(true){
        cin >> numero_casos;
        cin.ignore();
        if(numero_casos == 0)
            break;
        mg = 0;
        for(int i = 0; i < numero_casos; i++){
            getline(cin, frase);
            pos = frase.find(" ");
            numero = stoi(frase.substr(0, pos));
            fruta = frase.substr(pos+1);
            if(fruta == "suco de laranja")
                mg += numero*120;

            else if(fruta == "morango fresco")
                mg += numero*85;

            else if(fruta == "mamao")
                mg += numero*85;

            else if(fruta == "goiaba vermelha")
                mg += numero*70;

            else if(fruta == "manga")
                mg += numero*56;

            else if(fruta == "laranja")
                mg += numero*50;

            else if(fruta == "brocolis")
                mg += numero*34;
        }
        if(mg > 130)
            cout << "Menos "<< mg-130 << " mg\n";
        else if(mg >= 110)
            cout << mg << " mg\n";
        else
            cout << "Mais " << 110-mg << " mg\n";
    }
    return 0;        
}
